flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointExecuteResult.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+/**
+ * The result of executing a checkpoint.
+ */
+public class CheckpointExecuteResult {
+
+	private final CompletedCheckpoint completedCheckpoint;
+
+	private final CheckpointAbortReason failure;
+
+	private final Throwable failureCause;
+
+	CheckpointExecuteResult(CompletedCheckpoint success) {
+		this.completedCheckpoint = success;
```

Comment:
Instead of code-duplicating the assignments in both constructors, I suggest one private constructior that receives all 3 parameters, and the existing two will delegate to the new private constructor.